### PR TITLE
feat: Do not CSE Nat literals

### PR DESCRIPTION
**Pull request by bollu**
This fixes the performance scaling problems reported by @shigoel , where the tactic was performing too much work.

### CODE REVIEW

File: Tactics/CSE.lean [hunk 1]
```diff
@@ -154,14 +154,24 @@ private def Array.replicate (n : Nat) (v : α) : Array α := List.replicate n v
 def ExprData.isProfitable? (data : ExprData) : CSEM Bool :=
   return data.size > 1 && data.occs >= (← getConfig).minOccsToCSE
 
+/-- Check if an expression is a nat literal, or a `OfNat.ofNat` of a nat literal.
+This lets us avoid CSEing over nat literals, which are already in canonical form.
+-/
+def CSEM.isNatLit (e : Expr) : Bool :=
```
Comment (shigoel) [inline Tactics/CSE.lean:160]: Should other literals (e.g., `BitVec`) also deserve the same treatment as `Nat`s here?

Comment (bollu) [reply Tactics/CSE.lean:160]: @shigoel Good point. I'd say morally, yes. I feel that can go in the next PR?

Comment (shigoel) [reply Tactics/CSE.lean:160]: Agreed, counting on you to keep track! :)

**Review by shigoel — APPROVED** [review]
Thanks, @bollu! Looks great!

**Pull request merged**